File: evidence/generated-systems/angular/healthcare-on-angular/src/app/models/Appointment.js

// Define collection and schema for Appointment
export interface Appointment {
    appointmentDate:
	type : Date
    reason:
	type : string
    Patient:
	type : Schema.Types.ObjectId
    Clinician:
	type : Schema.Types.ObjectId
    Facility:
	type : Schema.Types.ObjectId
    Encounter:
	type : Schema.Types.ObjectId
    Status:
 	type : String
    Priority:
 	type : String
#
    collection: 'appointments'
}

File: evidence/generated-systems/golang/healthcare-on-golang/healthcare-on-golang/internal/model/Appointment.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// Appointment Declaration
//==============================================================
type Appointment struct {
    gorm.Model
     AppointmentDate                                                            time.Time
    Reason                                    string
    PatientId         *uint
    Patient           *Patient `gorm:"foreignKey:PatientId"`
    ClinicianId         *uint
    Clinician           *Clinician `gorm:"foreignKey:ClinicianId"`
    FacilityId         *uint
    Facility           *Facility `gorm:"foreignKey:FacilityId"`
    EncounterId         *uint
    Encounter           *Encounter `gorm:"foreignKey:EncounterId"`
    Status                      AppointmentStatus
    Priority                      Priority

// parent associations as their child

}

